### ISSUE TITLE
Draft/Publish issue

### ISSUE THREAD

**Issue by thezenmonkey** [bluehousegroup/silverstripe-data-object-version-viewer#11]
I'm using Gridfield Better Buttons, but I find that if I add the Version Viewer to the dataobject the 'Save Draft' buttom writes to live even though it shows as Draft int eh history.

**Comment by irkeepin** [comment]
Thanks for reporting this.

In our own work with this module, we ran into the same issue (plus other related issues). Our investigation revealed that the framework's implementation of Versioned on DataObjects is half-baked at best. BetterButtons goes some way to rectify this, but, to fully compensate for the wackiness in the core and get VersionViewer to work, we did have to tweak BetterButtons locally.

Others have had luck using this versioned-dataobjects module:

https://github.com/heyday/silverstripe-versioneddataobjects

It seems that this module compensates more fully for the framework's current lack of consistency on versioned DataObjects. In fact, we are currently looking into implementing this module ourselves, hopefully removing our need to modify BetterButtons locally.

Not trying to pass the buck! But there is only so much we can do while the framework itself remains inconsistent on this issue. I understand that there is an overhaul to DataObject versioning coming in 4.0 - cold comfort now, but at least it shows that the core developers recognize this issue and are working on it.

Also, for what it's worth, I highly recommend this silverstripe-admin module when doing any work with versioning in SS:

http://addons.silverstripe.org/add-ons/gdmedia/silverstripe-frontend-admin

It really helps keep straight which version of the site you're viewing, and makes it easy to switch between them. It's especially crucial if you have any clients or non-technical users playing around with versioning.
